gtest/src/check.rs

```diff
@@ -123,15 +125,52 @@ fn check_messages(
             messages.len(),
         ))
     } else {
+        let mut expected_messages: Vec<sample::Message> = expected_messages.into();
+        let mut messages: Vec<Message> = messages.into();
+
         expected_messages
-            .iter()
-            .zip(messages.iter())
+            .iter_mut()
+            .zip(messages.iter_mut())
             .enumerate()
             .for_each(|(position, (exp, msg))| {
+                let init = exp.init.unwrap_or(false);
                 if exp
                     .payload
-                    .as_ref()
-                    .map(|payload| !payload.equals(msg.payload.as_ref()))
+                    .as_mut()
+                    .map(|payload| match payload {
+                        PayloadVariant::Custom(_) => {
+                            for v in progs_n_paths.clone() {
+                                if ProgramId::from(v.1) == msg.source() {
+                                    let json = String::from_utf8(payload.to_bytes()).unwrap();
+
+                                    let bytes = if init {
+                                        js::get_bytes(&v.0, MetaType::InitOutput, json)
+                                    } else {
+                                        js::get_bytes(&v.0, MetaType::Output, json)
+                                    };
```

Comment:
Avoid repeating code in loops!

```suggestion
                                    let bytes = js::get_bytes(&v.0, meta_type, json);
```